### PR TITLE
feat(python): Raise an error when users try to use Polars API in a fork()-without-execve() child

### PR DESCRIPTION
**Pull request by itamarst**
This is an alternative implementation for #5342, which was reverted due to side-effects.

This implementation should not affect any non-Polars using code. It won't catch all cases of people using Polars in a fork()-without-execve() child, but it will catch many of them.

### CODE REVIEW

File: py-polars/polars/__init__.py [hunk 1]
```diff
@@ -414,3 +414,38 @@ def __getattr__(name: str) -> Any:
 
     msg = f"module {__name__!r} has no attribute {name!r}"
     raise AttributeError(msg)
+
+
+# fork() breaks Polars thread pool. Instead of silently hanging when users do
+# this, e.g. by using multiprocessing's footgun default setting on Linux, warn
+# them instead:
+def __install_postfork_hook() -> None:
```
Comment (ritchie46) [inline py-polars/polars/__init__.py:422]: Nice fix. Any idea of the startup cost of this?

Comment (itamarst) [reply py-polars/polars/__init__.py:422]: I suspect it's not meaningfully slower, but I'll measure it tomorrow.

Comment (itamarst) [reply py-polars/polars/__init__.py:422]: This appears to add 3 or 4 µs to `import polars`. This doesn't seem like a problem to me.

Comment (ritchie46) [reply py-polars/polars/__init__.py:422]: Alright. Thanks @itamarst. Nice that we can catch those deadlocks now.

**Pull request merged**